python/samseg/lta.py

```diff
@@ -103,49 +115,188 @@ def read(self, filename):
+                                      ]))
+
     def transformLines(self):
-        return [' '.join([str(col) for col in row]) for row in self.xform]
+        return [nice_array_format(row) for row in self.xform]
+
+    def calculate(self):
+        pass
 
 
-class LTA_MRI:
+class MRI:
+    def __init__(self):
+        self.analyzehdr = None
+        self.bhdr = None
+        self.valid = False
+
+    def read_header(self, filename):
+        self.fspec = filename
+        self.pwd = os.getcwd()
+        [self.vol, M, mr_parms, volsz] = load_mgh_header(filename)
+        [self.tr, self.flip_angle, self.te, self.ti] = mr_parms
+        volsz[0], volsz[1] = volsz[1], volsz[0]  # first two dimensions get swapped
+        self.vox2ras0 = M
+        self.volsize = volsz[0:3]
+
+        self.nframes = volsz[3]
+        # %--------------------------------------------------------------------%
+        # % Everything below is redundant in that they can be derivied from
+        # % stuff above, but they are in the MRI struct defined in mri.h, so I
+        # % thought I would add them here for completeness.  Note: MRIwrite()
+        # % derives all geometry information (ie, direction cosines, voxel
+        # % resolution, and P0 from vox2ras0. If you change other geometry
+        # % elements below, it will not be reflected in the output volume.
+        #
+        M2 = M * M
+
+        self.volres = [
+            math.sqrt(sum(M2[:, 0])),  # Column
+            math.sqrt(sum(M2[:, 1])),  # Row
+            math.sqrt(sum(M2[:, 2])),  # Slice
+        ]
+        self.xras = [M[row, 0] / self.xsize for row in range(3)]
+        self.yras = [M[row, 1] / self.xsize for row in range(3)]
+        self.zras = [M[row, 2] / self.xsize for row in range(3)]
+        ic = np.array([self.width / 2, self.height / 2, self.depth / 2, 1]).T
+        c = M @ ic
+        self.cras = list(c[0:3])
+        self.valid = True
+        return self
+
     def read(self, content):
         next(content)  # "xxx volume info"
         self.valid = parse_expected_int('valid', content)
         filename = parse_expected_string('filename', content)
-        self.volume = parse_expected_int_array('volume', content)
-        self.voxelsize = parse_expected_float_array('voxelsize', content)
+        self.volsize = parse_expected_int_array('volume', content)
+        self.volres = parse_expected_float_array('voxelsize', content)
         self.xras = parse_expected_float_array('xras', content)
         self.yras = parse_expected_float_array('yras', content)
         self.zras = parse_expected_float_array('zras', content)
         self.cras = parse_expected_float_array('cras', content)
+        self.upate_vox2ras()
         return filename
 
+    def upate_vox2ras(self):
+        # Compute the vox2ras matrix
+        Mdc = np.zeros([3, 3])
+        for row, source in enumerate([self.xras, self.yras, self.zras]):
```

Comment:
This can be replaced with an array broadcast
`Mdc[:, :] = np.array([[self.xras, self.yras, self.zras]])` or `Mdc[:, :] = np.array([[self.xras, self.yras, self.zras]]).T`